docs/resources/azurerm_virtual_machine.md.erb

```diff
@@ -0,0 +1,172 @@
+---
```

Comment:
At the moment it was just to avoid an implementation collision between the existing azure_virtual_machine resource and this implementation. I can update the docs to make that more clear, or we could rename all the new resources to the `azurerm` namespace if that would be more clear.

`azurerm` seems the preferred namespace in Terraform.

Reply:
azure vs azurerm is a common thing. It denotes the separation between Azure classic resource (Azure Service Manager or ASM) vs Azure Resource Manager (ARM)